holoviews/core/ndmapping.py

```diff
@@ -778,10 +779,8 @@ def label(self):
         if self._label:
             return self._label
         else:
-            labels = {v.label for v in self.values()
-                      if not v._auxiliary_component}
-            if len(labels) == 1:
-                return list(labels)[0]
+            if len(self):
+                return next(itervalues(self.data)).label
```

Comment:
My understanding is that label should be the empty string if the first value (i.e the one we are inspecting) is an ``_auxiliary_component``. Before we were iterating over the NdMapping elements: when you say the bottom layer shouldn't be an annotation, I believe you are thinking about the elements of an Overlay. In other words, I think we need this check for HoloMaps of annotations.